server/api/login.go

```diff
@@ -145,6 +145,30 @@ func HandleAuth(c *gin.Context) {
 		return
 	}
 
+	repos, err := _store.RepoListAll(true)
```

Comment:
well instead of itterate over all repos, wouldn't it be better to get the list a user has acces to from the forge?

in any case we need to be carefull as souch a large-scale-sync do hit api rate limits fast!!!

Reply:
> well instead of itterate over all repos, wouldn't it be better to get the list a user has acces to from the forge?

right - switched to that way of handling it, makes it much simpler too.

> in any case we need to be carefull as souch a large-scale-sync do hit api rate limits fast!!!

yeah about that, the `user_synced` attribute was dropped recently. https://github.com/woodpecker-ci/woodpecker/pull/1658/files and i couldn't find a good other attribute this could tie to.
Should we reintroduce it?

Reply:
well in this regards I think we should just encourage to not logout and in all the time ... the session can stay logged in quite some time D:

Reply:
fair enough - imho it would be nice if these were refreshed without a re-log (we have _a lot_ of traffic on our ci)
but thats outside of the scope of this PR anyway so  ¯\_(ツ)_/¯